### PR TITLE
Feature: Integrate WebPack in Bundle

### PR DESCRIPTION
**Pull request by ceesvanegmond**
## Title
Integrate WebPack in this bundle. Remove all compiled stuff and build them your self.

## Changelog

```markdown
### Added
- Added webpack config file

### Removed
- gulpfile.js and gulp dependencies
```

## To do

- [x] Update the documentation
- [x] Add an upgrade note

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    plugins: [
+        new CleanWebpackPlugin(['public/build/*']),
+        new Webpack.LoaderOptionsPlugin({
+            minimize: true,
+            debug: false,
+        }),
+        new Webpack.optimize.UglifyJsPlugin({
+            compress: true,
+            sourceMap: false
+        }),
```
Comment (Emroni) [inline webpack.config.js]: We've noticed that using the minimize and compress here results in differences in results sometimes. Rather remove lines 60-67.

File: package.json [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
+        "babel-preset-es2015-ie": "^6.7.0",
+        "magnific-popup": "^1.1.0",
+        "node-sass": "^4.5.2",
+        "postcss-loader": "^2.0.5",
+        "sass-loader": "^6.0.5",
+        "style-loader": "^0.17.0",
+        "trace.js": "^2.0.0",
+        "webpack": "^2.4.1"
+    }
 }
```
Comment (Emroni) [inline package.json]: No devDependencies?

**Review by Emroni — APPROVED** [review]

**Review by johnhout — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (Emroni) [reply package.json]: One other thing: both wearejust/trace and trace.js can be removed from the dependencies.

**Review by Emroni — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 1 above)
Comment (ceesvanegmond) [reply webpack.config.js]: Also from devDependencies? I quess ```        "@wearejust/trace": "^2.0.5",
``` must be in there right?

File: package.json (hunk 2 above)
Comment (ceesvanegmond) [reply package.json]: Also from devDependencies? I quess ```"@wearejust/trace": "^2.0.5",``` must be in there right?

File: webpack.config.js (hunk 1 above)
Comment (ceesvanegmond) [reply webpack.config.js]: @Emroni No minimize at all?

File: package.json (hunk 2 above)
Comment (Emroni) [reply package.json]: Sure, can be. Looks good to me :)

**Review by Emroni — DISMISSED** [review]

**Comment by ceesvanegmond** [comment]
@Emroni @johnhout I will target new 2.x branch since this is an BC.

**Pull request merged**